core/common/src/main/scala/org/typelevel/otel4s/baggage/BaggageManager.scala

```diff
@@ -30,29 +33,55 @@ import cats.mtl.Local
   *     _  <- Console[F].println("user_id: " + id)
   *   } yield ()
   *
-  * BaggageManager[F].local(io)(_.updated("user_id", "uid"))
+  * BaggageManager[F].modifiedScope(_.updated("user_id", "uid"))(io)
   *   }}}
   */
-trait BaggageManager[F[_]] extends Local[F, Baggage] {
+trait BaggageManager[F[_]] {
+
+  // TODO: remove after finishing migrating away from Local
+  protected[BaggageManager] def applicative: Applicative[F]
 
-  /** @return the current `Baggage` */
-  final def current: F[Baggage] = ask[Baggage]
+  /** @return the current scope's `Baggage` */
+  def current: F[Baggage]
 
   /** @return
     *   the [[Baggage.Entry entry]] to which the specified key is mapped, or `None` if the current `Baggage` contains no
     *   mapping for the key
     */
-  def get(key: String): F[Option[Baggage.Entry]] =
-    reader(_.get(key))
+  def get(key: String): F[Option[Baggage.Entry]]
 
   /** @return
     *   the value (without [[Baggage.Metadata metadata]]) to which the specified key is mapped, or `None` if the current
     *   `Baggage` contains no mapping for the key
     */
-  def getValue(key: String): F[Option[String]] =
-    reader(_.get(key).map(_.value))
+  def getValue(key: String): F[Option[String]]
+
+  /** Creates a new scope in which a modified version of the current `Baggage` is used. */
+  def modifiedScope[A](modify: Baggage => Baggage)(fa: F[A]): F[A]
+
+  /** Creates a new scope in which the given `Baggage` is used. */
+  def replacedScope[A](baggage: Baggage)(fa: F[A]): F[A]
```

Comment:
I'm not super happy with these names and am happy to bikeshed

Reply:
my concern with using the names `scope` and `local` is the following:

- the current methods have the effect as the second parameter list, which aligns with the `Tracer#<type>Scope` methods (and also is more pleasant to use IMO)
- the `Local` methods have the effect as the first parameter list

if we use the same names as `Local`, we have the choice of having a confusing API because it's not consistent with that of `Local`, or because it's not consistent with `Tracer` (and has a worse UX).

idk, maybe we just have to make that choice?

Reply:
I have been thinking about this, and I almost have thoughts, but I will wait until work hours to try to flesh them out